### PR TITLE
Handle get chat token 429

### PR DESCRIPTION
**Pull request by xTEddie**
- Update `getchattoken` to not retry on `429` failures
- Improve telemetry on `getchattoken` 
- Adjust `getchattoken` failures threshold

### CODE REVIEW

File: src/Utils/axiosRetry.ts [hunk 1]
```diff
@@ -8,6 +11,13 @@ import IAxiosRetryOptions from "../Interfaces/IAxiosRetryOptions";
  * @param axiosRetryOptions Options for axios retry.
  */
 const axiosRetry = (axios: AxiosInstance, axiosRetryOptions: IAxiosRetryOptions) => { // eslint-disable-line @typescript-eslint/explicit-module-boundary-types
+  const retryInterval = 1000; // 1 second interval between retries
+
+  // Default values
+  if (axiosRetryOptions.retryOn429 === undefined || axiosRetryOptions.retryOn429 === null) {
+    axiosRetryOptions.retryOn429 = true;
```
Comment (elopezanaya) [inline src/Utils/axiosRetry.ts:18]: I think default should be false, only true if it is explicitly defined

File: src/SDK.ts [hunk 2]
```diff
@@ -42,8 +42,9 @@ import { CustomContextData } from "./Utils/CustomContextData";
 
 export default class SDK implements ISDK {
   private static defaultConfiguration: ISDKConfiguration = {
-    getChatTokenRetryCount: 35,
+    getChatTokenRetryCount: 10,
```
Comment (elopezanaya) [inline src/SDK.ts:45]: was this number as result of an agreement or part of any discussion ?

File: src/SDK.ts [hunk 3]
```diff
@@ -42,8 +42,9 @@ import { CustomContextData } from "./Utils/CustomContextData";
...
     getChatTokenTimeBetweenRetriesOnFailure: 10000,
+    getChatTokenRetryOn429: false,
```
Comment (elopezanaya) [inline src/SDK.ts:47]: are we exposing this config to allow the customer set on-off ?or it will be hardcoded as well?,

File: src/SDK.ts (hunk 2 above)
Comment (elopezanaya) [inline src/SDK.ts:45]: did we set on not allow the customer to tune this values as he wants  via config json in lcw?

**Review by elopezanaya — CHANGES_REQUESTED** [review]

File: src/Utils/axiosRetry.ts (hunk 1 above)
Comment (xTEddie) [reply src/Utils/axiosRetry.ts:18]: `axiosRetry` is being used on every HTTP calls. No retry on 429 is only applicable on `getchattoken`. So, `retryOn429` should be `true` by default to keep backward compatibility.

File: src/SDK.ts (hunk 2 above)
Comment (xTEddie) [reply src/SDK.ts:45]: Based on investigation, we will not reach `35` tries. We reduce the value as a preemptive measure.

Comment (xTEddie) [reply src/SDK.ts:45]: Based on investigation, we will not reach 35 tries. We reduce the value as a preemptive measure.

Comment (xTEddie) [reply src/SDK.ts:45]: This value is abstracted from customer already. LCW has a config to explicitly see this value which is not exposed to customer.

**Pull request merged**